lib/Doctrine/ORM/Cache/Persister/Collection/AbstractCollectionPersister.php

```diff
@@ -264,4 +263,42 @@ public function loadCriteria(PersistentCollection $collection, Criteria $criteri
     {
         return $this->persister->loadCriteria($collection, $criteria);
     }
+
+    /**
+     * Clears cache entries related to the current collection
+     *
+     * @param PersistentCollection $collection
+     */
+    protected function evictCollectionCache(PersistentCollection $collection)
```

Comment:
I figured that none of the `EXTRA_LAZY` stuff was actually updating the collection-related caches. This change fixes that